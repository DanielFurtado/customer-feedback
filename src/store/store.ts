import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
