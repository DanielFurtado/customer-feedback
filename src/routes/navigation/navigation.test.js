import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import Navigation from './navigation';

describe(('Feedback Results'), () => {
  it('renders', () => {
    let component = renderer.create(
      <Provider store={store}>
        <Router>
          <Navigation />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct label and percentage', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navigation />
        </Router>
      </Provider>
    );
    expect(screen.getByRole('banner')).toBeInTheDocument();
    expect(screen.getByRole('menubar')).toBeInTheDocument();
  });
});