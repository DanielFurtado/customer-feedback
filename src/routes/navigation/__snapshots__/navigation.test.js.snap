// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Results renders 1`] = `
<header
  className="sc-bcXHqe gAzLXc"
  role="banner"
>
  <nav
    className="sc-gswNZR cYZNu"
  >
    <a
      aria-current="page"
      className="Nav--logo-container active"
      href="/"
      onClick={[Function]}
    >
      <svg>
        logo.svg
      </svg>
    </a>
    <ul
      className="sc-dkrFOg cybPPa"
      role="menubar"
    >
      <li
        role="none"
      >
        <a
          aria-current="page"
          className="Nav--link active"
          href="/"
          onClick={[Function]}
          role="menuitem"
        >
          Form
        </a>
      </li>
      <li
        role="none"
      >
        <a
          className="Nav--link"
          href="/results"
          onClick={[Function]}
          role="menuitem"
        >
          Results
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
